### PR TITLE
Migration StandaloneTranslationDoc to SQL (2 of 3)

### PR DESCRIPTION
**Pull request by orangejenny**
Branched off of https://github.com/dimagi/commcare-hq/pull/26702, first new commit is marked PR2.

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-1048)

### CODE REVIEW

File: corehq/apps/sms/views.py [hunk 1]
```diff
@@ -1639,46 +1638,46 @@ def edit_sms_languages(request, domain):
...
-                tdoc.translations[new] = tdoc.translations[old]
-                del tdoc.translations[old]
+                sms_translations.translations[new] = sms_translations.translations[old]
+                del sms_translations.translations[old]
 
         for lang in langs:
-            if lang not in tdoc.translations:
-                tdoc.translations[lang] = {}
+            if lang not in sms_translations.translations:
+                sms_translations.translations[lang] = {}
```
Comment (sravfeyn) [inline corehq/apps/sms/views.py]: `nit`: Could remove the `if` line by  doing `sms_translations.translations[lang] = sms_translations.translations.get(lang, {})`

File: corehq/apps/sms/views.py [hunk 2]
```diff
@@ -1697,14 +1696,15 @@ def upload_sms_translations(request, domain):
...
             msg_ids = sorted(_MESSAGES.keys())
             result = {}
-            for lang in tdoc.langs:
+            for lang in sms_translations.langs:
                 result[lang] = {}
 
             for row in translations:
-                for lang in tdoc.langs:
+                for lang in sms_translations.langs:
                     if row.get(lang):
```
Comment (sravfeyn) [inline corehq/apps/sms/views.py]: I know this comes from before you, but could reduce the deep nesting below by combining the if statement above with the one below.

File: corehq/apps/translations/models.py [hunk 3]
```diff
@@ -21,19 +23,37 @@
...
-class StandaloneTranslationDoc(TranslationMixin, CouchDocLockableMixIn):
+class SMSTranslations(SyncSQLToCouchMixin, models.Model):
+    domain = models.CharField(max_length=255, unique=True)
+    langs = JSONField(default=list)
+    translations = JSONField(default=dict)
+    couch_id = models.CharField(max_length=126, null=True, db_index=True)
+
+    @property
+    def default_lang(self):
+        if len(self.langs) > 0:
```
Comment (sravfeyn) [inline corehq/apps/translations/models.py]: Four lines to one `return self.langs[0] if self.langs else None`?

**Review by sravfeyn — APPROVED** [review]
Looks good, added some minor feedback

File: corehq/apps/sms/views.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/sms/views.py]: b488996fa36

File: corehq/apps/translations/models.py (hunk 3 above)
Comment (orangejenny) [reply corehq/apps/translations/models.py]: b488996fa36

File: corehq/apps/sms/views.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/sms/views.py]: leaving this one alone, the suggested code is shorter but I think counterintuitive to read

**Review by millerdev — APPROVED** [review]

**Pull request merged**